src/macros.rs

```diff
@@ -0,0 +1,23 @@
+#[macro_export]
+macro_rules! increment_counter {
+    ($state:ident.$property:ident, $metric:ident) => {{
+        use {opentelemetry::Context, tracing::debug};
+
+        if let Some(metrics) = &$state.$property {
+            metrics.$metric.add(&Context::current(), 1, &[]);
+            debug!("incremented `{}` counter", stringify!($metric));
+        }
+    }};
+}
```

Comment:
thought: I was going to suggest using a function instead of a macro here, but looks like the problem with that is not being able to read the metric name from the metric object itself. Interesting limitation.